packages/elements/src/rating/index.ts

```diff
@@ -42,123 +51,246 @@ export class Rating extends BasicElement {
   public interactive = false;
 
   /**
-    * Set number of total stars
-    */
+   * Internal max value of rating.
+   * Controlled by public setter and getter
+   */
+  private _max = '5';
+
+  /**
+   * Set number of total stars
+   * @param max max value
+   * @default '5'
+   */
   @property({ type: String })
-  public max = '5';
+  public set max (max: string) {
+    const newMax = max && this.isValidValue(max) ? Math.round(Number(max)).toString() : '5';
```

Comment:
Would 
```suggestion
    const newMax = max && this.isValidValue(max) ? Math.round(Number(max)).toString() : this.MAX_VALUE;
```
be better ?

Reply:
updated